projects/self_feeding/README.md

```diff
@@ -1,6 +1,112 @@
+```
+
+### Train on DIALOGUE (HH) + DIALOGUE (HB) examples
+To train on both HH and HB DIALOGUE examples, point the model to a train file that includes examples from both sets. For example, if you combined 131k HH DIALOGUE examples and 60k HB dialogue examples into a file called `train_hh131k_hb60k.txt`, you could add the following flag to train on that combined file for the DIALOGUE task:
+```
+--dia-train train_hh131k_hb60k.txt
+```
+
+### Train on DIALOGUE (HH) + FEEDBACK examples
+To train on more than one task (such as DIALOGUE and FEEDBACK), modify the command for training on DIALOGUE (HH) alone as follows:
+- Change `-t self_feeding:dialog:train` to `-t self_feeding:diafee:train`. This will result in a different "teacher" agent being used to train the chatbot, one with access to both 'dia\[logue\]' and 'fee\[dback\]`.
+- Append `,fee_acc,fee_loss` to the end of the `-tbmetrics` argument to include the accuracy and loss terms for the feedback task to the Tensorboard logs.
+- Add `--fee-weight 1.0` to set a coefficient by which the loss for the `feedback` task will be multiplied (the default is 1.0 already).
+
+Putting this all together, the command to recreate the 131k HH + 60k FB result from the paper is as follows (as reported in Table 9 in the paper, this setting had the same optimal hyperparameter settings as 131k HH):
+```
+python -u $PARLAIHOME/examples/train_model.py -t self_feeding:diafee:train --model-file $MODEL -tblog true -tbcomment $MODEL -tbmetrics lr,dia_acc,dia_loss,dia_rank,fee_acc,fee_loss -ltim 5 -vtim 10 -vp 10 -m projects.self_feeding.self_feeding_agent:SelfFeedingAgent -cands batch --eval-candidates inline -histsz 2 --embedding-type fasttext_cc --embedding-size 300 --dict-maxtokens 250000 --num-epochs 100 --optimizer adamax --embeddings-scale false -bs 128 --relu-dropout 0 --attention-dropout 0 --n-heads 2 --n-layers 2 -lr 0.0025 --ffn-size 32 --lr-scheduler invsqrt --warmup-updates 500 -vmt dia_acc -vmm max
+```
+
+### Train on DIALOGUE (HH) + DIALOGUE (HB) + FEEDBACK (FB) + SATISFACTION (ST) examples
+You can train on all three tasks at once with the command below.
+```
+python -u $PARLAIHOME/examples/train_model.py -t self_feeding:all:train --model-file $MODEL -tblog true -tbcomment $MODEL -tbmetrics lr,dia_acc,dia_loss,dia_rank,fee_acc,fee_loss,sat_acc,sat_f1,sat_loss -ltim 5 -vtim 10 -vp 50 -m projects.self_feeding.self_feeding_agent:SelfFeedingAgent -cands batch --eval-candidates inline -histsz 2 --embedding-type fasttext_cc --embedding-size 300 --dict-maxtokens 250000 --num-epochs 500 --optimizer adamax --embeddings-scale false -bs 128 --relu-dropout 0 --attention-dropout 0 --n-heads 2 --n-layers 2 -lr 0.0025 --ffn-size 32 --lr-scheduler invsqrt --warmup-updates 500 --dia-train train_hh131k_hb60k.txt -vmt dia_acc -vmm max
```

Comment:
this failed for me. :(

```python-traceback
Traceback (most recent call last):
  File "examples/train_model.py", line 16, in <module>
    TrainLoop(opt).train()
  File "/private/home/roller/working/parlai/parlai/scripts/train_model.py", line 361, in __init__
    build_dict(opt, skip_if_built=True)
  File "/private/home/roller/working/parlai/parlai/scripts/build_dict.py", line 118, in build_dict
    world_dict = create_task(ordered_opt, dictionary)
  File "/private/home/roller/working/parlai/parlai/core/worlds.py", line 1156, in create_task
    world = create_task_world(opt, user_agents, default_world=default_world)
  File "/private/home/roller/working/parlai/parlai/core/worlds.py", line 1113, in create_task_world
    opt, user_agents, default_world=default_world
  File "/private/home/roller/working/parlai/parlai/core/worlds.py", line 1068, in _get_task_world
    task_agents = _create_task_agents(opt)
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 823, in _create_task_agents
    return create_task_agent_from_taskname(opt)
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 776, in create_task_agent_from_taskname
    task_agents = teacher_class(opt)
  File "/private/home/roller/working/parlai/parlai/tasks/self_feeding/agents.py", line 355, in __init__
    super().__init__(opt, shared)
  File "/private/home/roller/working/parlai/parlai/tasks/self_feeding/agents.py", line 218, in __init__
    super().__init__(opt, shared)
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 230, in __init__
    self.tasks.extend(create_task_agent_from_taskname(opt_singletask))
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 776, in create_task_agent_from_taskname
    task_agents = teacher_class(opt)
  File "/private/home/roller/working/parlai/parlai/tasks/self_feeding/agents.py", line 276, in __init__
    super().__init__(opt, shared)
  File "/private/home/roller/working/parlai/parlai/tasks/self_feeding/agents.py", line 74, in __init__
    raise ValueError("Unrecognized filepath: {}".format(path))
ValueError: Unrecognized filepath: /private/home/roller/working/parlai/data/self_feeding/self_feeding_v02/train_hh131k_hb60k.txt
```
```
devfair0237 parlai patch-1 » ls /private/home/roller/working/parlai/data/self_feeding/self_feeding_v02
convai2_cands.txt  test_hh.txt  train_fb_a.txt  train_fb.txt  train_hh_hb.txt  train_st.txt  valid_hh.txt
test_fb.txt        test_st.txt  train_fb_b.txt  train_hb.txt  train_hh.txt     valid_fb.txt  valid_st.txt
```

Reply:
Discussed offline. Result: the particular derivative file used in this command will be provided in the updated dataset version (rather than requiring the user to construct it themselves by concatenating the right number of lines from the corresponding files).